### ISSUE TITLE
recent path search not working for new paths

### ISSUE THREAD

**Issue by ajilty** [marlonrichert/zsh-autocomplete#752]
## Environment

```zsh
% typeset -p1 VENDOR OSTYPE ZSH_PATCHLEVEL _autocomplete__funcfiletrace
typeset VENDOR=apple
typeset OSTYPE=darwin23.0
typeset ZSH_PATCHLEVEL=zsh-5.9-0-g73d3173
typeset -a _autocomplete__funcfiletrace=(
  /var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/zsh-autocomplete/zsh-autocomplete.plugin.zsh:4
  /var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/.zshrc:9
  zsh:0

% git -C ~autocomplete log --oneline -n1
762afac (grafted, HEAD -> main, tag: 24.09.04, origin/main, origin/HEAD) Add note to Readme about usage of `%`

```

* Operating system: macOS 14.6.1 (23G93)
* Terminal emulator: Terminal.app

## Steps to reproduce

```zsh
% cd $(mktemp -d)
% git clone --depth 1 -- https://github.com/marlonrichert/zsh-autocomplete.git
Cloning into 'zsh-autocomplete'...
remote: Enumerating objects: 53, done.
remote: Counting objects: 100% (53/53), done.
remote: Compressing objects: 100% (51/51), done.
remote: Total 53 (delta 0), reused 31 (delta 0), pack-reused 0 (from 0)
Receiving objects: 100% (53/53), 1.64 MiB | 13.75 MiB/s, done.

% > .zshrc <<EOF
setopt interactivecomments transientrprompt
PS1='%# '
PS2=
RPS2='%^'

source /var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/zsh-autocomplete/zsh-autocomplete.plugin.zsh

autoload -Uz chpwd_recent_dirs cdr
add-zsh-hook chpwd chpwd_recent_dirs

EOF
% env -i HOME=$PWD PATH=$PATH TERM=$TERM ${TERMINFO:+TERMINFO=$TERMINFO} zsh -d
% mkdir -p foo/bar/yummy  
% cd foo/bar 
% cd yummy
% cd ~
% cdr -l
1    /private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/foo/bar/yummy
2    /private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/foo/bar
3    /private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi

% tail ~/.local/share/zsh/chpwd-recent-dirs
$'/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi'
$'/private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/foo/bar/yummy'
$'/private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/foo/bar'
$'/private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi'

% +autocomplete:recent-directories
% echo $reply
/private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/foo/bar/yummy /private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/foo/bar /private/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi


# type `yummy`, hit tab, no suggestions
# up arrow does do the normal things

```

<details>
<summary>Contents of <code>/var/folders/j6/pjs4p22d77v7bvs6ds534c140000gp/T/tmp.gmBn94rnKi/.local/state/zsh-autocomplete/log/2024-09-12.log</code> (click to expand)</summary>
<pre>
(empty)
</pre>
</details>

**Comment by Rayzeq** [comment]
I have the same issue
I'm on Nixos with zsh-5.9-0-g73d3173.
Completions for recent directories works with 23.07.13 but not with 24.09.04.

**Comment by faileon** [comment]
I guess the latest release changed the behaviour,
> https://github.com/marlonrichert/zsh-autocomplete/commit/a8fdd21227d76e077ad034c6ff7108ae44bd57a7 Move recent paths completion to a keybinding. Resolves https://github.com/marlonrichert/zsh-autocomplete/issues/640.

Is there a way to get the old one back? My autocompletions now no longer suggest the paths I use the most.
for example I could write
`$ cd abc`
and the autocomplete would suggest
`~/Projects/company/abc`
which was super neat

**Comment by faileon** [comment]
@marlonrichert do you have any suggestions please?

**Comment by tommeier** [comment]
ah this is what broke `recent-directories` - I thought it was upgrading to sequoia - is downgrading the version in homebrew possible 🤔

**Comment by tommeier** [comment]
For those that miss autocomplete on `~/Projects/company/abc` as much as me 😅 

```
brew uninstall zsh-autocomplete
brew tap homebrew/core --force
brew extract --version=23.07.13 zsh-autocomplete $USER/local-zsh-autocomplete
brew install zsh-autocomplete@23.07.13
brew pin zsh-autocomplete@23.07.13
```

You'll then need to update your `source`, eg: `source /path/to/zsh-autocomplete/zsh-autocomplete.plugin.zsh` -> `source /path/to/zsh-autocomplete@23.07.13/zsh-autocomplete.plugin.zsh` (for me `/path/to` == `brew --prefix`) and then autocomplete recent-directories working as before 🎉

**Comment by faileon** [comment]
Downgrading is surely a possible workaround. Hopefully it won't be the only solution and we can get this behavior back to the new versions :]

**Comment by runofthemill** [comment]
trying to figure this out, and upon looking at https://github.com/marlonrichert/zsh-autocomplete/issues/761 I figured trying `setopt interactivecomments` - and that seems to work! 

looking at the [issue template](https://github.com/marlonrichert/zsh-autocomplete/blob/762afacbf227ecd173e899d10a28a478b4c84a3f/.github/ISSUE_TEMPLATE/bug-report.md?plain=1#L32), `setopt interactivecomments` is part of the boilerplate, so seems like the new release changed requires `interactivecomments` to be set. 

**tl;dr** - docs probably just need to be updated to note `setopt interactivecomments` must be set

**Comment by faileon** [comment]
> trying to figure this out, and upon looking at #761 I figured trying `setopt interactivecomments` - and that seems to work!
> 
> looking at the [issue template](https://github.com/marlonrichert/zsh-autocomplete/blob/762afacbf227ecd173e899d10a28a478b4c84a3f/.github/ISSUE_TEMPLATE/bug-report.md?plain=1#L32), `setopt interactivecomments` is part of the boilerplate, so seems like the new release changed requires `interactivecomments` to be set.
> 
> **tl;dr** - docs probably just need to be updated to note `setopt interactivecomments` must be set

Hmm it seems to not work on my end. my .zshrc has the following:
```
setopt interactivecomments transientrprompt
```
but the recent paths still do not work. Really sad I can't get it working other way than downgrading, it was a great feature :(

**Comment by ttytm** [comment]
The recent path completion has been completely broken for a few months. I've been holding on to an earlier commit ever since. Cd into your zsh-autocomplete plugin/submodule path and 
```
git checkout 2912c9788c08bbb86106ebd35dd94a376a2b2e99
```

**Comment by marlonrichert** [comment]
Recent paths search has been moved to a separate [keyboard shortcut](https://github.com/marlonrichert/zsh-autocomplete?tab=readme-ov-file#keyboard-shortcuts), as it often interfered with normal path completion or vice versa. You can now toggle it on and off by pressing `^X/`.

**Comment by Rayzeq** [comment]
Are there any plans to add an option to revert to the old behaviour ? I really preferred it to the new one 😔.

**Comment by faileon** [comment]
> Are there any plans to add an option to revert to the old behaviour ? I really preferred it to the new one 😔.

Same here, having a shortcut kind of defeats the purpose of autocompletion for something i do thousand times a day. It's much easier to type `cd abc` hit tab and have it autocomplete to the first matching directory I use the most, i.e. `cd ~/Projects/company/abc`.

**Comment by marlonrichert** [comment]
I found workaround. Add this to your `.zshrc` file to include recent directories whenever directories are listed, except when the current word is empty or equal to the path to an existing directory:
```zsh
zstyle -e ':completion:*:directories' fake '[[ -z $PREFIX$SUFFIX || -d $PREFIX$SUFFIX ]] || +autocomplete:recent-directories'
```
